src/lib/ppx_coda/check_ocaml_word_size.ml

```diff
@@ -0,0 +1,26 @@
+open Core_kernel
+open Ppxlib
+open Asttypes
+
+(** This is a ppx to check that we're building with an n-bits OCaml
+
+    Usage: [%%check_ocaml_word_size n]
+
+    There will be a compile-time error if the check fails
+ *)
+
+let name = "check_ocaml_word_size"
+
+let expand ~loc ~path:_ expected_word_size =
+  if Int.(not @@ equal Sys.word_size expected_word_size) then
```

Comment:
NIT: This threw me a bit; perhaps `not (Int.equal Sys.word_size expected_word_size)` would be clearer if it doesn't change the semantics.